cpp/include/cudf/io/parquet.hpp

```diff
@@ -951,140 +885,100 @@ class parquet_writer_options {
    *
    * @param comp_stats Pointer to compression statistics to be updated after writing
    */
-  void set_compression_statistics(std::shared_ptr<writer_compression_statistics> comp_stats)
-  {
-    _compression_stats = std::move(comp_stats);
-  }
+  void set_compression_statistics(std::shared_ptr<writer_compression_statistics> comp_stats);
 
   /**
    * @brief Sets preference for V2 page headers. Write V2 page headers if set to `true`.
    *
    * @param val Boolean value to enable/disable writing of V2 page headers.
    */
-  void enable_write_v2_headers(bool val) { _v2_page_headers = val; }
+  void enable_write_v2_headers(bool val);
 
   /**
    * @brief Sets sorting columns.
    *
    * @param sorting_columns Column sort order metadata
    */
-  void set_sorting_columns(std::vector<sorting_column> sorting_columns)
-  {
-    _sorting_columns = std::move(sorting_columns);
-  }
+  void set_sorting_columns(std::vector<sorting_column> sorting_columns);
 };
 
 /**
- * @brief Class to build `parquet_writer_options`.
+ * @brief Base class for Parquet options builders.
  */
-class parquet_writer_options_builder {
-  parquet_writer_options options;
+template <class BuilderT, class OptionsT>
+class parquet_writer_options_builder_base {
+  OptionsT _options;
 
- public:
+ protected:
   /**
-   * @brief Default constructor.
+   * @brief Return reference to the options object being built
    *
-   * This has been added since Cython requires a default constructor to create objects on stack.
+   * @return the options object
    */
-  explicit parquet_writer_options_builder() = default;
+  inline OptionsT& get_options() { return _options; }
 
   /**
-   * @brief Constructor from sink and table.
+   * @brief Constructor from options.
    *
-   * @param sink The sink used for writer output
-   * @param table Table to be written to output
+   * @param options Options object to build
    */
-  explicit parquet_writer_options_builder(sink_info const& sink, table_view const& table)
-    : options(sink, table)
-  {
-  }
+  explicit parquet_writer_options_builder_base(OptionsT options);
```

Comment:
should this be a const&?

Reply:
IIRC this was a clangd suggestion along with passing in an initializer list.